src/main/java/picocli/CommandLine.java

```diff
@@ -13932,6 +13964,10 @@ private void processPositionalParameter(Collection<ArgSpec> required, Set<ArgSpe
                 if (tracer.isDebug()) {
                     tracer.debug("Parser is configured to treat all unmatched options as positional parameter", arg);}
             }
+            if (!endOfOptions && !commandSpec.parser().parameterAllowedBeforeEndOfOptions()) {
+                handleUnmatchedArgument(args);
```

Comment:
Currently, it is `Unmatched argument at index 5: 'a'` where the index is the position of the "bad" input and 'a' is the value of the bad input.  I assume that you would prefer a new message, perhaps like `Unmatched argument at index 5: 'a'.  Positional parameters must follow the "{EndOfOptions Character, like --}"`

Reply:
Yes please!

How about:

`Unmatched argument at index 5: 'a'.  Positional parameters must follow the EndOfOptions delimiter '--'.`

(The delimiter string is configurable, so the code that generates this error message should use `get EndOfOptionsDelimiter` instead of hardcoding the default `--`)

Reply:
This change is a bit tricky due to the existing code flow.  Please take a close look at this part of the change before approving.